metrics: Fix enabling pmlogger in Metrics Settings if pcp is not installed

martinpitt opened the pull request:
Previously this just failed with "unit pmlogger.service does not exist".
Instead, when enabling pmlogger, offer to install the package. Just like
with the button in the empty state, request a relogin afterwards to pick
up the new manifest.

martinpitt commented:
In the integration test we only have a fake package. But I interactively tested that with the real Fedora packages, and it DTRT: After removing pcp, enabling pmlogger in cockpit, logging out and back in, history metrics appear automatically.

KKoukiou commented on pkg/metrics/metrics.jsx:
```diff
@@ -905,13 +905,27 @@ const PCPConfig = ({ buttonVariant, firewalldRequest }) => {
+                        setNeedsLogout(true);
+                        handleSave();
+                    })
+                    .catch(() => null); // ignore cancel
```
This will however ignore all exceptions not just cancelled promise, right?

martinpitt replied:
Right, but the dialog already handles package install failures etc. internally (as inline alerts). `install_dialog()` itself is documented accordingly:

> The install_dialog function returns a promise that is fulfilled when the dialog closes after
> a successful installation.  The promise is rejected when the user cancels the dialog.

We use it the same way in pkg/packagekit/autoupdates.jsx or pkg/storaged/optional-panel.jsx.

But indeed I just saw that we don't handle cancel() in handleInstall() on the metrics page -- this causes an unhandled exception that appears in the log. I'll fix that in a separate commit.

KKoukiou replied:
ack thanks

martinpitt commented:
There's some test failures (and in addition, that confusing situation where two tests run in parallel!), the missing cancel handling of the already existing install dialog, and a "servicie" typo in a comment, back to draft.

martinpitt commented:
Indeed the logic to not recursively trigger the install dialog [was broken](https://logs.cockpit-project.org/logs/pull-15998-20210625-125357-183ec17b-debian-testing/log.html#272). Fixed now, and fixed another race condition in the test.

martinpitt commented:
There was still a race condition in the Install → apply Save handover, which I noticed in the on-demand redis installation. This ought to work better, and extend to other packages.

martinpitt commented:
Oops, this [causes some unexpected fallout](https://logs.cockpit-project.org/logs/pull-15998-20210628-105155-dd17683f-centos-8-stream/log.html#147), I suppose the test should better stop pmlogger in the end.

martinpitt commented:
@KKoukiou : Ready to review for real now.
